x11rb/src/rust_connection/stream.rs

```diff
@@ -351,6 +351,18 @@ fn do_write(
 
     let res = if !fds.is_empty() {
         let fds = fds.iter().map(|fd| fd.as_fd()).collect::<Vec<_>>();
+
+        // Most sendmsg implementations put a limit of at least 0xFF file descriptors.
+        if fds.len() > 0xFF {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                format!(
+                    "Cannot send more than 0xFF FDs at once, tried to send {}",
+                    fds.len()
+                ),
+            ));
+        }
```

Comment:
Why do we have to check this manually? Shouldn't `sendmsg()` simply fail if we try to send too many FDs without having to check the limit before-hand?

I checked libxcb: It has an internal `#define XCB_MAX_PASS_FD 16`. So 255 is way more than that. XCB's code also doesn't really say where that limit comes from (but they need one since they have a static allocation and not a growable vector for saving FDs).

Reply:
Removed this manual check